v2/internal/util/match/literal_matcher.go

```diff
@@ -3,47 +3,76 @@
  * Licensed under the MIT license.
  */
 
-package config
+package match
 
 import (
 	"strings"
+
+	"github.com/Azure/azure-service-operator/v2/internal/util/typo"
 )
 
 // literalMatcher is a StringMatcher that provides a case-insensitive match against a given string
 type literalMatcher struct {
 	literal string
 	matched bool
-	advisor *TypoAdvisor
+	advisor *typo.Advisor
 }
 
 var _ StringMatcher = &literalMatcher{}
 
 // newLiteralMatcher returns a new literalMatcher for the given string
-func newLiteralMatcher(literal string) *literalMatcher {
+func newLiteralMatcher(literal string) StringMatcher {
 	return &literalMatcher{
 		literal: strings.TrimSpace(literal),
-		advisor: NewTypoAdvisor(),
+		advisor: typo.NewAdvisor(),
+	}
+}
+
+func (lm *literalMatcher) Matches(value string) Result {
+	if strings.EqualFold(lm.literal, strings.TrimSpace(value)) {
+		if !lm.matched {
+			// First time we match, clear out our advisory as we won't be using it
+			lm.matched = true
+			lm.advisor.ClearTerms()
+		}
+
+		return Result{
+			Matches:         true,
+			MatchingPattern: lm.literal,
+		}
+	}
+
+	if !lm.matched {
+		// Still collecting potential suggestions
+		lm.advisor.AddTerm(value)
+	}
+
+	return Result{
+		Matches:         false,
+		MatchingPattern: "",
 	}
 }
 
-// Matches returns true if the passed value is a case-insensitive match with our configured literal
-func (lm *literalMatcher) Matches(value string) bool {
+// MatchesDetailed returns true if the provided value is matched by the matcher
+// If there was a match, the second return value contains the pattern that matched.
+// If there was not a match, the second return value contains ""
+func (lm *literalMatcher) MatchesDetailed(value string) (bool, string) {
```

Comment:
Isn't `Matches()` now already returning this information?

Reply:
Doh, yes. I missed deleting this method, good catch